### ISSUE TITLE
RecordSpec.forBaseUrl(...) and recordSpec().forTarget(...).build() behave differently

### ISSUE THREAD

**Issue by bhelfert** [tomakehurst/wiremock#1419]
Hello and thank you very much for developing and providing WireMock!

**tl;dr**

Is it a defect that `RecordSpec.forBaseUrl("http://someurl.com")` and `WireMock.recordSpec().forTarget("http://someurl.com").build()` result in a different behavior or is that intended?

**Context and details**

While recording JIRA REST API responses with `wiremock-jre8:2.27.2` and using the (somewhat old) [Spock Wiremock Extension](https://github.com/felipefzdz/spock-wiremock-extension) I have run into the issue that some responses are not recorded. (As we will see soon these responses have a body with a text length > 10 KiB.)

When using WireMock's library API directly (i.e. instead of the Spock Wiremock Extension) that issue did not appear.

Comparing the code of the Spock Wiremock Extension (1) and the one I have used (2) I discovered that:
- [server.startRecording(recordSpec().forTarget("http://someurl.com"))](https://github.com/felipefzdz/spock-wiremock-extension/blob/master/src/main/java/com/felipefzdz/spock/WiremockScenarioInterceptor.java#L159) from (1)
- [server.startRecording("http://someurl.com")](https://github.com/tomakehurst/wiremock/blob/2.27.2/src/main/java/com/github/tomakehurst/wiremock/core/WireMockApp.java#L451) from (2)

behave differently, what I did not expect.

The reason is that in case (1) the `RecordSpecBuilder.build()` method creates a new `RecordSpec` object that [has a configured ResponseDefinitionBodyMatcher](https://github.com/tomakehurst/wiremock/blob/2.27.2/src/main/java/com/github/tomakehurst/wiremock/recording/RecordSpecBuilder.java#L147) as `extractBodyCriteria`. This matcher limits responses having a [text length greater than 10 KiB](https://github.com/tomakehurst/wiremock/blob/2.27.2/src/main/java/com/github/tomakehurst/wiremock/recording/ResponseDefinitionBodyMatcher.java#L31) to get extracted to files after recording.

Case (2) does not configure this `ResponseDefinitionBodyMatcher`.

Is it intended that these two library API calls behave differently or is that a defect?

**Comment by rprzystasz** [comment]
Seems like logic mismatch caused by multiple usages of `RecordSpec` constructor.
The only place it should be used is the `build()` method inside Builder.
Additionally constructor could be package private to disallow incorrect initialisation. 

The only concern is that existing tests with usages of `RecordSpec::forBaseUrl` might start failing.
